components/brave_vpn/resources/panel/components/select-region-list/index.tsx

```diff
@@ -2,97 +2,112 @@
-import { getLocale } from '../../../../../common/locale'
-import Flag from '../flag'
-
-function SelectRegion () {
-  // TODO(nullhook): Scroll to the selected radio input when this component loads
-  // TODO(nullhook): Add search region functionality
-  const dispatch = useDispatch()
-  const currentRegion = useSelector(state => state.currentRegion)
-  const regions = useSelector(state => state.regions)
-  const [selectedRegion, setSelectedRegion] = React.useState(currentRegion)
-  const ref = React.useRef() as React.MutableRefObject<HTMLButtonElement>
+import { getLocale } from '$web-common/locale'
 
-  const handleGoBackClick = () => {
-    dispatch(Actions.toggleRegionSelector(false))
-  }
+import 'emptykit.css'
 
-  const handleItemClick = (currentRegion: Region) => {
-    // Using a local state will help to keep the selection
-    // consistent, if the user exits without pressing connect
-    setSelectedRegion(currentRegion)
-  }
+interface RegionContentProps {
+  region: Region
+  selected: boolean
+}
 
+function RegionContent(props: RegionContentProps) {
+  const dispatch = useDispatch()
   const handleConnect = () => {
-    dispatch(Actions.connectToNewRegion(selectedRegion))
+    dispatch(Actions.connectToNewRegion(props.region))
   }
 
-  if (!selectedRegion) {
-    console.error('Selected region is not defined')
-  }
+  const ref = React.useRef<HTMLDivElement>(null)
 
   React.useEffect(() => {
-    if (ref.current && currentRegion) {
-      ref.current.scrollIntoView()
-    }
+    if (props.selected) ref.current?.scrollIntoView()
   }, [])
 
+  return (
+    <S.RegionContainer
+      selected={props.selected}
+      ref={ref}
+    >
+      <S.RegionCountry selected={props.selected}>
+        <Flag countryCode={props.region.countryIsoCode} />
+        <S.RegionCountryLabel id='country-name'>
+          {props.region.namePretty}
+        </S.RegionCountryLabel>
+        {props.selected && (
+          <S.StyledCheckBox name='check-circle-filled'></S.StyledCheckBox>
+        )}
+        <S.RegionConnect
+          slot='actions'
+          kind='filled'
+          size='tiny'
+          onClick={handleConnect}
+          id='connect'
+        >
+          {getLocale('braveVpnConnect')}
+        </S.RegionConnect>
+      </S.RegionCountry>
+    </S.RegionContainer>
+  )
+}
+
+export function PanelHeader(props: {
+  title: string
+  buttonAriaLabel: string
+  onClick: () => void
+}) {
+  return (
+    <S.PanelHeader>
+      <S.StyledButton
+        type='button'
+        onClick={props.onClick}
+        aria-label={props.buttonAriaLabel}
+      >
+        <S.StyledIcon name='arrow-left'></S.StyledIcon>
+      </S.StyledButton>
+      <S.HeaderLabel>{props.title}</S.HeaderLabel>
+    </S.PanelHeader>
+  )
+}
+
+function SelectRegion() {
+  const dispatch = useDispatch()
+  const currentRegion = useSelector((state) => state.currentRegion)
+  const regions = useSelector((state) => state.regions)
+
+  const handleGoBackClick = () => {
```

Comment:
nit: function doesn't depend on component state so we can lift it out of the component (which should reduce unnecessary rerenders

Reply:
Oh, I missed that `dispatch` was a hook :facepalm:  - sorry @simonhong